src/spatial/inertia.hpp

```diff
@@ -266,6 +268,42 @@ namespace pinocchio
       return M;
     }
 
+    /** Returns the representation of the matrix as a vector of dynamic parameters.
+    * The parameters are given as
+    * \f$ v = [m, mc_x, mc_y, mc_z, I_{xx}, I_{xy}, I_{yy}, I_{xz}, I_{yz}, I_{zz}]^T \f$
+    * where \f$ I = I_C + mS^T(c)S(c) \f$ and \f$ I_C \f$ has its origin at the barycenter.
+    */
+    Vector10 toDynamicParameters() const
+    {
+      Vector10 v;
+
+      v[0] = mass();
+      v.template segment<3>(1) = mass() * lever();
```

Comment:
Use noalias here

Reply:
Wait. Really, noalias? That's a product between a scalar and a vector. Is noalias needed?

Reply:
It will do allocation for it.

Reply:
I am surprised, I thought a scalar product would not do it. However, it costs nothing to add it.

Reply:
I think Eigen does not consider aliasing of the form `a.segment<1,2>() = a.segment<0,2>();`. So there is no aliasing the case mentioned above.

Moreover, noalias for vectors of very small size is counter-productive. It can be *more efficient* to use a temporary.

Reply:
Yes, I do not see where the temporary allocation might arise.

Reply:
> I think Eigen does not consider aliasing of the form `a.segment<1,2>() = a.segment<0,2>();`. So there is no aliasing the case mentioned above.
> 
> Moreover, noalias for vectors of very small size is counter-productive. It can be _more efficient_ to use a temporary.

By default, Eigen is considering aliasing for any product expression.

Reply:
> > I think Eigen does not consider aliasing of the form `a.segment<1,2>() = a.segment<0,2>();`. So there is no aliasing the case mentioned above.
> > Moreover, noalias for vectors of very small size is counter-productive. It can be _more efficient_ to use a temporary.
> 
> By default, Eigen is considering aliasing for any product expression. Indeed, it has no clue to guest it at compile time.

Any *matrix* product. This is a *scalar* product. Eigen knows it and there is no reasoin to fear aliasing. From Eigen's doc:

[https://eigen.tuxfamily.org/dox/group__TopicAliasing.html](https://eigen.tuxfamily.org/dox/group__TopicAliasing.html)
> Aliasing is harmless with coefficient-wise computations; **this includes scalar multiplication** and matrix or array addition.
**When you multiply two matrices, Eigen assumes that aliasing occurs.** If you know that there is no aliasing, then you can use noalias().
**In all other situations, Eigen assumes that there is no aliasing issue** and thus gives the wrong result if aliasing does in fact occur. To prevent this, you have to use eval() or one of the xxxInPlace() functions.

(emphasis mine)

I will remove the noalias() as it is useless

Reply:
> By default, Eigen is considering aliasing for any product expression.

for any matrix-matrix product, not scalar-matrix product, which falls into unary operation in Eigen.

Reply:
Ok, sounds good to me then. I was not aware of it before.

Reply:
I removed it